Updates scripts/add-dates.js to handle more edge cases for adding/formatting dates in tables, ensuring 2-space padding (e.g., |  2025-07-02  |).

BankkRoll opened the pull request:
## Changes
- **Handle Missing `|`**: Append `|  ${currentDate}  |` to rows with enough columns but no final `|`.
- **Invalid Dates**: Replace non-`YYYY-MM-DD` or invalid dates with current date.
- **Malformed Rows**: Skip rows with too few columns or invalid Markdown, log warnings.
- **Broken Links**: Ignored, dates added regardless.
- **Logging**: Reduced to essentials (changes made, malformed rows, file status).
- **Error Handling**: Exit on file read/write errors.

## Edge Cases & Logs
```markdown
# Input README.md
| Name | Description | Link | Date |
|------|-------------|------|------|
| test1 | Desc1 | [Link](https://test1.com) |     |             # Empty date
| test2 | Desc2 | [Link](https://test2.com) | 2024-12-06 |      # Valid date
| test3 | Desc3 | [Link](https://test3.com) | 2024/12/06 |      # Invalid date
| test4 | Desc4 | [Link](invalid) |     |                       # Broken link
| test5 | Desc5 | [Link](https://test5.com) |                   # Missing final `|`
| test6 | Desc6 | [Link](https://test6.com) |  2024-12-06  |    # Valid, correct spacing
| test7 | Desc7 | [Link](https://test7.com) | invalid |         # Invalid date
| test8 | Desc8 | [Link](https://test8.com) | 2024-13-01 |      # Invalid date
| test9 | Desc9 |                                               # Too few columns
| test10 | Desc10 | [Link](https://test10.com) | 2024-12-06 | extra |  # Extra column

# Output README.md
| Name | Description | Link | Date |
|------|-------------|------|------|
| test1 | Desc1 | [Link](https://test1.com) |  2025-07-02  |
| test2 | Desc2 | [Link](https://test2.com) |  2024-12-06  |
| test3 | Desc3 | [Link](https://test3.com) |  2025-07-02  |
| test4 | Desc4 | [Link](invalid) |  2025-07-02  |
| test5 | Desc5 | [Link](https://test5.com) |  2025-07-02  |
| test6 | Desc6 | [Link](https://test6.com) |  2024-12-06  |
| test7 | Desc7 | [Link](https://test7.com) |  2025-07-02  |
| test8 | Desc8 | [Link](https://test8.com) |  2025-07-02  |
| test9 | Desc9 |                                       
| test10 | Desc10 | [Link](https://test10.com) |  2024-12-06  | extra |


# Console Logs
Processing README.md with 13 lines
Skipping malformed row at line 11: | test9 | Desc9 |
Skipping malformed row at line 12: | | | | |
Skipping malformed row at line 13: |broken
Completed processing. Made 7 changes to date columns.
Encountered 3 malformed or misaligned rows.
Changes written to README.md

birobirobiro commented:
Hey, @BankkRoll  you're still working on this script together with the other repository, right?

BankkRoll commented:
> Hey, @BankkRoll you're still working on this script together with the other repository, right?

Hey, no i seen you updated the formatting, as the actions bot was messing up formatting when adding dates. This PR is just updating the actions bot script to use correct formatting when updating the dates on a PR and some additional date format checks.

--

The other repo for the site, assume you are confused with this script [init-dates.js](https://github.com/birobirobiro/awesome-shadcn-ui-site/blob/main/scripts/init-dates.js), i should possibly delete that as it was just the initial actions bot set up when i added it i needed to fetch all past PRs to get their dates initially so that scipt is now unused... but i just converted the site to app router should have removed it

BankkRoll commented:
> > Hey, no i seen you updated the formatting on some of the date here in the readme, as the actions bot was messing up formatting when adding dates. This PR is just updating the actions bot script to use correct formatting when updating the dates on a PR and some additional date format checks.
> > --
> > The other repo for the site, assume you are confused with this script [init-dates.js](https://github.com/birobirobiro/awesome-shadcn-ui-site/blob/main/scripts/init-dates.js), i should possibly delete that as it was just the initial actions bot set up when i added it i needed to fetch all past PRs to get their dates initially so that script is now unused... but i just converted the site to app router should have removed it
> 
> Can I accept this PR from now on?

Yes! This pr will fix the actions bot to format better

BankkRoll commented:
see there was a bug already fixing

birobirobiro commented:
> see there was a bug already fixing

Perfect, I realized that two columns of dates are being created.
